p2p_wallet/Scenes/Main/History/History.TransactionParser.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Created by Giang Long Tran on 15.04.2022.
+//
+
+import Foundation
+import SolanaSwift
+
+protocol HistoryTransactionParser {
+    ///  Parse transaction.
+    ///
+    /// - Parameters:
+    ///   - signatureInfo: the raw signature info
+    ///   - transactionInfo: the raw transaction info
+    ///   - account: the account that the transaction info belongs to.
+    ///   - symbol: the token symbol that the transaction has to do.
+    /// - Returns: parsed transaction
+    func parse(
+        signatureInfo: SolanaSDK.SignatureInfo,
+        transactionInfo: SolanaSDK.TransactionInfo,
+        account: String,
+        symbol: String
+    ) async throws -> SolanaSDK.ParsedTransaction
+}
+
+extension History {
+    typealias TransactionParser = HistoryTransactionParser
```

Comment:
typealias isn't needed hear. Use typealiases if you have many nested types as for example `HistorySomething.Element.Lol.etc`

Reply:
I agree, I will drop all typealiases